include/irr/asset/filters/dithering/CWhiteNoiseDither.h

```diff
@@ -21,20 +19,46 @@ namespace irr
 				class CState
 				{
 					public:
-						CState() {}
+						CState()
+							: sampler(std::chrono::high_resolution_clock::now().time_since_epoch().count())
+						{
+						
+						}
 						virtual ~CState() {}
+
+						core::RandomSampler sampler;
```

Comment:
nope, you dont want a random number generator to implement this... you want a hash function and feed it the `pixelCoord` and `channel` as input to permute into a single 32 bit value (that you then divide by ~0u to get a float in the correct range)

Reply:
solved